src/main/java/com/xpandit/plugins/xrayjenkins/Utils/ProxyUtil.java

```diff
@@ -31,13 +37,38 @@ private ProxyUtil() {}
         if (proxyConfiguration != null) {
             final HttpHost proxy = getProxy(proxyConfiguration);
             final CredentialsProvider credentialsProvider = getCredentialsProvider(proxyConfiguration);
-            
-            return new HttpRequestProvider.ProxyBean(proxy, credentialsProvider, proxyConfiguration.getNoProxyHostPatterns());
+            final List<Pattern> noProxyUrlPatterns = getNoProxyUrlPatterns(proxyConfiguration.noProxyHost);
+
+            return new HttpRequestProvider.ProxyBean(proxy, credentialsProvider, noProxyUrlPatterns);
         }
-        
+
         return null;
     }
 
+    // Similar to hudson.ProxyConfiguration.getNoProxyHostPatterns(String), but returns URL patterns, instead of domain patterns.
+    private static List<Pattern> getNoProxyUrlPatterns(String noProxyHost) {
+        if (StringUtils.isBlank(noProxyHost)) {
+            return Collections.emptyList();
+        }
+        return Stream.of(noProxyHost.split("[ \t\n,|]+"))
+            .filter(StringUtils::isNotEmpty)
+            .map(ProxyUtil::makeUrlPattern)
+            .collect(Collectors.toList());
+    }
+
+    private static Pattern makeUrlPattern(String noProxyHostElement) {
+        final String regexp;
+        if (noProxyHostElement.indexOf("://") >= 0) {
+            // looks like an URL pattern; it's not supported in Jenkins in general, but it was in previous versions
+            // of this plugin (2.3.0, 2.3.1), so let's keep that unchanged in case someone ever used it successfuly...
+            regexp = noProxyHostElement.replace(".", "\\.").replace("*", ".*");
```

Comment:
Probably I am missing something, but why aren't you ending this pattern with _"(:[0-9]+)?(/.*)?"_ ?

This would allow for the same flexibility provided in the _else_ condition below.

Reply:
> Probably I am missing something, but why aren't you ending this pattern with "(:[0-9]+)?(/.*)?" ?
> This would allow for the same flexibility provided in the else condition below.

This case is there only for backward compatibility, for people who have already added an URL pattern (like "`https://jira.something.com/*`") in their proxy exclusions list because that's what was necessary with XRay plugin 2.3.0 or 2.3.1. The minimal pattern=>regexp conversion it does is the same `hudson.ProxyConfiguration.getNoProxyHostPatterns()` does, so if it was working before, then great, it will still work in later XRay plugin versions, despite not being meaningful for Jenkins core or other plugins.

But I don't want to encourage new variations of the URL patterns thing, because it is specific to the XRay plugin. If you add a "(:[0-9]+)?(/.*)?" suffix, then it becomes more likely that some users will try "`https://jira.something.com`" in their proxy exclusion list, will stick with that because it works for the XRay plugin, and then, when (for instance) they later try the Jira plugin, they would wonder why it is using the proxy instead of honouring their (apparently perfectly fine) exclusion config.

So I think it's better to avoid additional flexibility here, to stick to what's required for backward compatibility, and to mainly support the "right" syntax instead (hosts names patterns, so "`jira.something.com`" in this case).

Reply:
Agreed. Thank you for your explanation!